File: 30-days-of-code-challenges/day_21_generics.java
import java.lang.reflect.Method;

class day_21_generics {
    //Write your code here
    public static <E> void printArray (E[] inputArray) {
        for (E element: inputArray) {
            System.out.println(element);
        }
    }
	
    public static void main(String args[]){
        Integer[] intArray = { 1, 2, 3 };
        String[] stringArray = { "Hello", "World" };
        
        printArray( intArray  );
        printArray( stringArray );
        
        if(Solution.class.getDeclaredMethods().length > 2){
            System.out.println("You should only have 1 method named printArray.");
        }
    }
}

File: 30-days-of-code-challenges/day_21_generics.cpp
#include <iostream>
#include <vector>

using namespace std;

// Add your code here
template<typename T> 
void printArray(vector<T> array) {
     for(T i : array){
          cout << i << endl;
     }
}

int main() {
  
    vector<int> vInt{1, 2, 3};
    vector<string> vString{"Hello", "World"};
    
    printArray<int>(vInt);
    printArray<string>(vString);
    
    return 0;
}